server/server.js

```diff
@@ -1,16 +1,84 @@
 import express from 'express';
 import 'dotenv/config'
 import db from "./db/db-connection.js";
+import cors from 'cors';
 
 const app = express();
 const PORT = 1212;
 
+app.use(cors());
+app.use(express.json()); //req.body
+
+app.get("/", async (req, res) =>  {
+    
+    try {
+        const {rows : allContacts} = await db.query('SELECT * FROM contacts');
+        res.send(allContacts);
+        console.log(allContacts)
+    } catch (error) {
+        console.error("Error Message!:", error.message);
+    }
 
-app.get("/", (req, res) => {
-    res.json("Welcome to the Contact App");
 });
 
-const test = await db.query('SELECT * FROM contacts')
-console.log("table: ", test)
+app.post("/", async (req, res) =>  {
+    
+    try {
+        const {first_name, last_name, email, phone_number, address, image_path } = req.body;
+
+        const newContact = await db.query (
+            "INSERT INTO contacts (first_name, last_name, email, phone_number, address, image_path) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *", [first_name, last_name, email, phone_number, address, image_path]
+        );
+
+        res.json(newContact.rows[0])
+        
+    } catch (error) {
+        console.error(error.message)
+    }
+
+});
+
+app.delete('/:id', async (req, res) => {
+    try {
+        const {id} = req.params;
+        const deleteContact= await db.query("DELETE FROM contacts WHERE id = $1", [id]
+        );
+
+        res.json("The contact was deleted!");
+
+    } catch (error) {
+        console.error(error.message)
+    }
+})
+
+app.put('/:id', async (req, res) => {
+
+    try {
+
+      const { id } = req.params;
+
+      const {first_name, last_name, email, phone_number, address, image_path } = req.body;
+
+      const editedContact = await db.query(
+        "UPDATE contacts SET first_name = $1, last_name = $2, email = $3, phone_number = $4, address = $5, image_path = $6 WHERE id = $7 RETURNING *",
+        [first_name, last_name, email, phone_number, address, image_path, id]
+      );
+
+        if (editedContact.rows.length > 0) {
+          
+            res.json({
+                status: 'success',
+                message: 'Contact updated!',
+                data: editedContact.rows[0]
+            });
+
+            console.log(editedContact.rows[0])
+        }
+
+    } catch(error){
+        console.log(error);
+        return res.status(400).json({error});
```

Comment:
```suggestion
        res.status(400).json({error});
```

This isn't something you can `return`. You just call it normally like you did above